dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataMetadataConnector.java

```diff
@@ -103,7 +104,9 @@ public void addTasksTo(List<? super Task<?>> out, ConnectorArguments arguments)
             .toWhereClause();
     String whereBDatabaseNameClause =
         new SqlBuilder()
-            .withWhereInVals("\"B_DatabaseName\"", arguments.getDatabases())
+            .withWhereInVals(
+                "UPPER(\"B_DatabaseName\")",
```

Comment:
This one field is case sensitive? fun

Reply:
It's a combination of `UPPERCASE` column attribute:

```
REPLACE VIEW DBC.AllTempTablesVX
AS SELECT
...
  Dbase.DatabaseNameI  (NAMED B_DatabaseName)
FROM DBC.Dbase ...
```

```
CREATE SET TABLE DBC.Dbase ...
     (
      DatabaseNameI VARCHAR(128) CHARACTER SET UNICODE UPPERCASE NOT CASESPECIFIC NOT NULL,
...
```
https://docs.teradata.com/r/Enterprise_IntelliFlex_VMware/SQL-Data-Definition-Language-Syntax-and-Examples/Table-Statements/ALTER-TABLE/ALTER-TABLE-Syntax-Elements/Column-Attributes

and the ANSI transaction session mode that JDBC driver uses instead of TERA mode:

https://github.com/google/dwh-migration-tools/blob/1edf19c945eab55ad5c7393f5e24118edcc0d98e/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/AbstractTeradataConnector.java#L169

ANSI mode is case-sensitive - it ignores the `NOT CASESPECIFIC` column attribute.

"ANSI does all character comparisons as the equivalent of CASESPECIFIC."
https://docs.teradata.com/r/SQL-Functions-Operators-Expressions-and-Predicates/June-2017/Comparison-Operators-and-Functions/Character-String-Comparisons/Case-Sensitivity